client/client-benchmark-noop-api-plugin/src/main/java/org/elasticsearch/plugin/noop/action/bulk/RestNoopBulkAction.java

```diff
@@ -32,24 +32,28 @@
 import org.elasticsearch.rest.BaseRestHandler;
 import org.elasticsearch.rest.BytesRestResponse;
 import org.elasticsearch.rest.RestChannel;
-import org.elasticsearch.rest.RestController;
 import org.elasticsearch.rest.RestRequest;
 import org.elasticsearch.rest.RestResponse;
 import org.elasticsearch.rest.action.RestBuilderListener;
 
 import java.io.IOException;
+import java.util.List;
 
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
 import static org.elasticsearch.rest.RestRequest.Method.POST;
 import static org.elasticsearch.rest.RestRequest.Method.PUT;
 import static org.elasticsearch.rest.RestStatus.OK;
 
 public class RestNoopBulkAction extends BaseRestHandler {
 
-    public RestNoopBulkAction(RestController controller) {
-        controller.registerHandler(POST, "/_noop_bulk", this);
-        controller.registerHandler(PUT, "/_noop_bulk", this);
-        controller.registerHandler(POST, "/{index}/_noop_bulk", this);
-        controller.registerHandler(PUT, "/{index}/_noop_bulk", this);
+    @Override
+    public List<Route> handledRoutes() {
```

Comment:
handlesRoutes?

Any chance you could reverse the order of the args on the routes ctor? In my mind the method comes first.

Reply:
On further thought, I lean more towards `routes` rather than `handledRoutes` or `handlesRoutes`. What do you think?

I agree regarding the parameter ordering.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/51950/commits/086a29595ffa8820814d925c64f97bc40b4fe741 to change the method name and https://github.com/elastic/elasticsearch/pull/51950/commits/265327123f4b59d7775f6ec62fe1622cf12b1bff to change the parameter ordering.